creativecloud/blocks/sidenav/sidenav.js

```diff
@@ -1,38 +1,97 @@
 import { createTag } from '../../scripts/utils.js';
-import { html, css, LitElement } from '/creativecloud/deps/lit-all.min.js';
+
+import('/creativecloud/deps/lit-all.min.js');
 import('/creativecloud/deps/merch-spectrum.min.js');
-import('../../deps/sidenav.js');
+import('/creativecloud/deps/merch-sidenav.js');
 
-const getValueFromLabel = (label) => label
+const getValueFromLabel = (content) => content
   .trim()
   .toLowerCase()
-  .replace(/(and|-)/g, '')
+  .replace(/( and )/g, ' ')
+  .replace(/-/g, '')
   .replace(/\s+/g, '-');
 
-const appendSidenavItem = (parent, li) => {
-  const label = li.childNodes[0]?.textContent;
-  const value = getValueFromLabel(label);
-  const item = createTag('sp-sidenav-item', { label, value });
-  parent.append(item);
-  const childList = li.querySelector('ul');
-  if (childList) {
-    childList.querySelectorAll('li').forEach((grandChild) => {
-      appendSidenavItem(item, grandChild);
-    });
+const getCategories = (arrayCategories) => {
+  const tag = createTag('merch-sidenav-list', { deeplink: 'category' });
+  const mapParents = {};
+  arrayCategories.forEach((item) => {
+    if (item.Name?.length > 0) {
+      const parentName = item.Name.split('|')[0].trim();
+      if (!mapParents[parentName]) {
+        mapParents[parentName] = createTag('sp-sidenav-item', { label: parentName, value: getValueFromLabel(parentName) });
+        tag.append(mapParents[parentName]);
+      }
+      const childName = item.Name.split('|')[1]?.trim();
+      if (childName) {
+        const childNode = createTag('sp-sidenav-item', { label: childName, value: getValueFromLabel(childName) });
+        mapParents[parentName].append(childNode);
+      }
+    }
+  });
+  return tag;
+};
+
+const getTypes = (arrayTypes) => {
+  const tag = createTag('merch-sidenav-checkbox-group', { title: 'Types', deeplink: 'types' });
+  arrayTypes.forEach((item) => {
+    if (item.Name?.length > 0) {
+      const checkbox = createTag('sp-checkbox', { emphasized: '', name: getValueFromLabel(item.Name) });
+      checkbox.append(document.createTextNode(item.Name));
+      tag.append(checkbox);
+    }
+  });
+  return tag;
+};
+
+const appendFilters = async (root, link) => {
+  const payload = link.textContent.trim();
+  const resp = await fetch(payload);
```

Comment:
fetch should be surrounded with try catch and failure should be gracefully handled.